### DISCUSSION TITLE
Use logarithmic scale for axis

### DISCUSSION THREAD

**Discussion by DominicD** [patrykandpatrick/vico#432] [Ideas]
Hi I would like to use a logarithmic scale for the y-axis of a chart. Is there a way to do that?

**Comment by patrickmichalik** [comment]
Hello! I’m afraid this isn’t supported at the moment, but it’s something that we’ll be looking into implementing. Thanks for the suggestion!

**Comment by DominicD** [comment]
@patrickmichalik Thank you for the fast response! 
Your library works nice! Maybe it would be nice to use the github project feature or somethin else to track the Roadmap? 
I think this would make the lib even more attractive for people that take a look.

**Comment by patrickmichalik** [comment]
Thanks for the kind words! Due to the project’s nature, we’re rather spontaneous with what we work on. One unpredictable variable is how much spare time we have. As a result, a detailed roadmap would, unfortunately, be hard for us to maintain. We’d not be able to provide more information than one can collect by taking a look at the issue and discussion lists.

**Comment by y9san9** [comment]
I already implemented such scale by myself and would like to contribute, @patrickmichalik what can you suggest to start from?

![IMG_20230926_110211_369](https://github.com/patrykandpatrick/vico/assets/46930374/e8e454d7-c254-4d8a-a974-f5b70deda23e)
![IMG_20230926_110213_420](https://github.com/patrykandpatrick/vico/assets/46930374/dd2e03cc-2b47-46ba-8da3-239858719806)

**Comment by patrickmichalik** [comment]
Thanks for your interest in contributing, @y9san9. Please feel free to open a pull request. We have several automatic checks, which you can run by using these Gradle tasks:

- `detekt` (code formatting)
- `testDebug` (unit tests)
- `verifyPaparazziDebug` (screenshot tests)

**Comment by kuti-JP** [comment]
Hi @y9san9 could you share the code to achieve this, or some directions on how you did this please?

**Reply by Tyler-Lopez** [reply]
If I wanted to do this, my approach would be to do the logarithmic conversion out of Vico so that Vico is receiving linear data, and then, in the value formatter for the labels, take the linear value and derive the correct logarithmic value.

This would require no change to Vico which I like

**Reply by patrickmichalik** [reply]
Thanks for the input, @Tyler-Lopez! Such an approach would work. The series would include the logarithms of the original _y_ values to the desired base. During formatting and the like, the base would be exponentiated.

In charts with a logarithmic _y_ scale, it’s sometimes desired for labels or lines to be displayed for _y_ values that aren’t equally spaced on the scale. Here, this could be achieved by means of a custom `VerticalAxis.ItemPlacer` implementation.
